aptos-move/mvhashmap/src/unsync_map.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{types::MVCodeOutput, utils::module_hash};
+use aptos_crypto::hash::HashValue;
+use aptos_types::{
+    executable::{Executable, ExecutableDescriptor, ModulePath},
+    write_set::TransactionWrite,
+};
+use std::{
+    cell::RefCell,
+    collections::{
+        btree_map::Entry::{Occupied, Vacant},
+        BTreeMap, HashMap,
+    },
+    hash::Hash,
+};
+
+struct Entry<V: TransactionWrite, X: Executable> {
+    // data is Option since we may just use the entry to cache base (storage-version) executable.
```

Comment:
Just making sure. Even if you only have the base at the end of the block you are going to store it in the executable_cache, right?

Reply:
executable cache will not be in the view, and it can exist for parallel or sequential.

Reply:
then we can decide to flush it after sequential (too much trouble for little gain to store latest executable for each path and sequentially update it, sequential is already slow), or update in parallel from MVModulesMap after the execution.